BusinessLogic/LootLogic.cs

```diff
@@ -269,39 +269,266 @@ public async Task<bool> SetLootRequestStatus(int playerId, int currentPlayerId,
             return true;
         }
 
-        public LootOverviewViewModel GetOverview(int[] display, int syndicateId)
+        public LootOverviewViewModel GetOverview(int[] display, int tierSelect, int typeSelect, int syndicateId)
         {
             var lootStatuses = EnumExtensions.ToDictionary<int>(typeof(CampaignLoot.CampaignLootStatus));
             if (display == null || display.Length == 0)
             {
                 display = lootStatuses.Keys.ToArray();
             }
-            var model = new LootOverviewViewModel{
-                AllLoot = _dbContext.CampaignLoot
-                    .Include(l => l.Campaign)
-                    .Include(l => l.Holder)
-                    .Include(l => l.Item)
-                    .Join(
-                        _dbContext.Campaign,
-                        cl => cl.CampaignId,
-                        c => c.Id,
-                        (cl, c) => new { CampaignLoot = cl, Campaign = c }
-                    )
-                    .Where(clc => clc.Campaign.SyndicateId == syndicateId)
-                    .Select(clc => clc.CampaignLoot)
-                    .Where(cl => display == null || display.Contains((int)cl.Status)).OrderByDescending(cl => cl.Campaign.UTCDateTime),
-                OtherSyndicatesLoot = _dbContext.CampaignLoot
-                    .Include(cl => cl.Campaign)
-                    .ThenInclude(clc => clc.Syndicate)
-                    .Where(cl => cl.Campaign.SyndicateId != syndicateId && 
-                        cl.AvailableToOtherSyndicates == true &&
-                        (cl.Status == CampaignLoot.CampaignLootStatus.Undistributed ||
-                        cl.Status == CampaignLoot.CampaignLootStatus.StaysWithSyndicate))
-                    .ToList(), // ToList() is, sadly, necessary to persist syndicate info/tag.
-                LootStatuses = EnumExtensions.ToDictionary<int>(typeof(CampaignLoot.CampaignLootStatus)),
-                Display = display,
-            };
-            return model;
+            // And again, I can't enter conditional clauses into those queries translated directly into SQL
+            // Gonna have some IFs
+            if (tierSelect > 0) { // Tier selected
+                if (typeSelect == 0) { // All types of items
+                    var model = new LootOverviewViewModel{
+                        AllLoot = _dbContext.CampaignLoot
+                            .Include(l => l.Campaign)
+                            .Include(l => l.Holder)
+                            .Include(l => l.Item)
+                            .Where(l => l.Item.Tier == tierSelect)
```

Comment:
For example, here you can write `.Where(l => l.Item.Tier == tierSelect || tierSelect == 0)`. This will eliminate the need for `if (tierSelect > 0)`. Under the hood, if condition under `Where()` can be evaluated even before making a request to the database, it WILL be evaluated first and will not even affect the SQL that Entity Framework generates. In this case it will evaluate to `true` if `tierSelect` equals 0 - and this would mean that there will be no additional condition passed to the DB call.